yarn-project/simulator/src/avm/journal/journal.ts

```diff
@@ -307,12 +307,14 @@ export class AvmPersistableStateManager {
     this.transitionalExecutionResult.allUnencryptedLogs.push(ulog);
     // this duplicates exactly what happens in the trace just for the purpose of transitional integration with the kernel
     this.transitionalExecutionResult.unencryptedLogsHashes.push(
-      new SideEffect(logHash, new Fr(this.trace.accessCounter)),
+      new LogHash(logHash, this.trace.accessCounter, new Fr(ulog.length)),
     );
     // Duplicates computation performed in public_context.nr::emit_unencrypted_log
     // 44 = addr (32) + selector (4) + raw log len (4) + processed log len (4).
-    this.transitionalExecutionResult.unencryptedLogPreimagesLength = new Fr(
-      this.transitionalExecutionResult.unencryptedLogPreimagesLength.toNumber() + 44 + log.length * Fr.SIZE_IN_BYTES,
+    // Note that ulog.length includes all the above bytes apart from processed log len
+    // Processed log len is added to replicate conversion to function_l2_logs at the end of exec.
+    this.transitionalExecutionResult.unencryptedLogPreimagesLength = new Fr(ulog.length + 4).add(
```

Comment:
@AVM team: This is just a cleanup - we don't need the +44 or `SIZE_IN_BYTES`, that was only in the circuit to replicate `log.length` as handled by the ts log class.
The extra +4 is to pre-handle conversion to `Function_L2_Logs` later on in the ts path, so the length covers all bytes the user should pay for in gas. I'm not sure how relevant this is for the AVM, but kept it so the values are the same.

Reply:
I see the +44 still in [PublicContext::finish()](https://github.com/AztecProtocol/aztec-packages/pull/6268/files#diff-fa0913141d1a8c03ac3c9ec1a4eda8bd6dbdbccc3e01ea5e82849f5ed76d6b81R283). Why does it need to be there but doesn't need to happen in the AVM?

Reply:
It does happen - in `ulog.length`. `ulog` is an `UnencryptedL2Log` class, and `.length` returns:
`EventSelector.SIZE + this.data.length + AztecAddress.SIZE_IN_BYTES + 4`
which is: `4 + this.data.length + 32 + 4 = this.data.length + 40`, hence the extra `+4` above.

Reply:
Got it, makes sense!